src/boundary_conditions.jl

```diff
@@ -31,46 +31,42 @@ struct CubicPeriodicBoundaryConditions{cType <: Real} <: BoundaryConditions
     L::cType
 end
 
-function apply_boundary_conditions!(ri, rj, pbc::PeriodicBoundaryConditions, R2)
-    rij = @MVector [Inf, Inf, Inf]
-    success = false
-    rij2 = 0
-
-    for dx in [0,-pbc[2],pbc[2]], dy in [0,-pbc[4],pbc[4]], dz in [0,-pbc[6],pbc[6]]
-        rij = @MVector [ri[1] - rj[1] + dx, ri[2] - rj[2] + dy, ri[3] - rj[3] + dz]
-        for x in (1, 2, 3) 
-            rij[x] -= (pbc[2x] - pbc[2x - 1]) * div(rij[x], (pbc[2x] - pbc[2x - 1]))
-        end
-        rij2 = dot(rij, rij)
-        if  rij2 < R2
-            success = true
-            break
-        end
-    end
-    return (rij, rij2, success)
+function get_interparticle_distance(ri, rj, pbc::PeriodicBoundaryConditions)
+    rij = ri - rj
+    r, r2 = zero(eltype(ri)), zero(eltype(ri))
+    x, y, z = rij
+    while x  < pbc[1]   x += pbc[2]-pbc[1] end
+    while x >= pbc[2]   x -= pbc[2]-pbc[1] end
+    while y  < pbc[3]   y += pbc[4]-pbc[3] end
+    while y >= pbc[4]   y -= pbc[4]-pbc[3] end
+    while z  < pbc[5]   z += pbc[6]-pbc[5] end
+    while z >= pbc[6]   z -= pbc[6]-pbc[5] end
+    rij = @SVector [x, y, z]
+    r2 = dot(rij, rij)
```

Comment:
does it not optimize it more if it knows that `rij` are both the same? I would think literal `x^2 + y^2 + z^2` would be faster?

Reply:
In my benchmarking the times are exactly the same, but there does seem to be an extraneous `vpermilpd` instruction (and the single precision equivalent for `Float32`s), so I'll expand it out just in case.